genai-perf/docs/embeddings.md

```diff
@@ -36,18 +36,18 @@ GenAI-Perf allows you to profile embedding models running on an
 To create a sample embeddings input file, use the following command:
 
 ```bash
-echo '{"text": "What was the first car ever driven?"}
-{"text": "Who served as the 5th President of the United States of America?"}
-{"text": "Is the Sydney Opera House located in Australia?"}
-{"text": "In what state did they film Shrek 2?"}' > embeddings.jsonl
+echo '{"text_input": "What was the first car ever driven?"}
```

Comment:
Doesn't this technically constitute a breaking change in the tool? What's the general policy regarding breaking changes in general?

Reply:
Great question. The tool is still in alpha, or unofficially beta. We have had talks to go GA. We unfortunately are still releasing some breaking changes regularly (e.g. changing CLI args), though we are trying to start getting that down to 0.

I'll need to discuss this with @debermudez and the team. We needed to publish embeddings and rankings support in about a week, and we ended up deviating the way the other endpoints support text versus this endpoint. We could modify the retrievers to support both text and text_input as equivalent fields if we wanted to include this. I think it'd still be good to update documentation to just use one approach so that we can eventually deprecate the other.

@debermudez @ganeshku1 @nv-hwoo @nicolasnoble What are your thoughts on supporting the use of both text and text_input as equivalent fields for now but updating the documentation to use text_input everywhere?

Reply:
Alternatively, text is shorter, so we could always support _text_ everywhere. I'm open. Just would prefer it be the same everywhere for clarity (and cleaner code later).

Reply:
The excuse of the tool not being GA is a perfectly valid one to change things like this, to be fair :-) I just don't know if we have an expectation of notifying users of changes like this one.

Reply:
We usually add breaking changes to the release notes. For now, I added support for both text and text_input, so those using the older format can continue doing so. If we want to deprecate one of the two, we can make that decision when we go GA.

Reply:
This feels like the right approach to me.
That way we arent breaking users for now, and gives us some flexibility

Reply:
> What are your thoughts on supporting the use of both text and text_input as equivalent fields for now but updating the documentation to use text_input everywhere?

I personally prefer `text` over `text_input` just because it's simpler and for better consistency with other inputs (e.g. users define image as `image`, not `image_input`).

Reply:
We can switch to text. I'd do that in its own PR, as changing `text_input` to `text` everywhere has a lot of points at which it can break code, so I want it to be individually reviewed.

Also, we should update parser.py and README at some point. It looks like --input-file needs to be updated to show that it accepts both text and image fields.